bittensor/_synapse/synapse.py

```diff
@@ -35,13 +35,22 @@ class SynapseCall( ABC ):
     def __init__(
         self,
         synapse: 'bittensor.Synapse',
-        request_proto: object
+        request_proto: object,
+        context: grpc.ServicerContext,
     ):
+        metadata = dict(context.invocation_metadata())
+        (
+            _,
+            sender_hotkey,
+            _,
+            _,
+        ) = synapse.axon.auth_interceptor.parse_signature(metadata)
+        
         self.completed = False
         self.start_time = time.time()
         self.timeout = request_proto.timeout
         self.src_version = request_proto.version
-        self.src_hotkey = request_proto.hotkey
+        self.src_hotkey = sender_hotkey
```

Comment:
This indeed prevents spoofing the source hotkey, but still allows the request to go through.

Do we want to check somewhere in the synapse that the request hotkey or still let the request pass?

e.g. in the apply() call:
```python
    def apply( self, call: SynapseCall ) -> object:
        ....
            if call.src_hotkey != call.request_hotkey:
                call.return_code = bittensor.proto.ReturnCode.InvalidRequestHotkey
                call.return_message = 'Invalid request hotkey'
                raise InvalidRequestHotkeyError
```

Reply:
The thing is that the `request_hotkey` (I am not sure this is the path in the object - the field is `hotkey` in the request proto) field should be removed completely. Older clients do not have this verification anyway, but do need it, while newer clients ignore it completely, and don't need the field - you are not preventing anything by raising the exception.

Having an extra return code for such cases is superflous and just pollutes the codebase with cases which should be completely removed in the end anyway.

Reply:
Agreed. Thanks for the explanation.